L1Trigger/L1THGCal/src/veryfrontend/HGCalVFELinearizationImpl.cc

```diff
@@ -1,15 +1,23 @@
 #include "L1Trigger/L1THGCal/interface/veryfrontend/HGCalVFELinearizationImpl.h"
 
 HGCalVFELinearizationImpl::HGCalVFELinearizationImpl(const edm::ParameterSet& conf)
-    : linLSB_(conf.getParameter<double>("linLSB")),
-      adcsaturation_(conf.getParameter<double>("adcsaturation")),
-      tdcnBits_(conf.getParameter<uint32_t>("tdcnBits")),
-      tdcOnsetfC_(conf.getParameter<double>("tdcOnsetfC")),
-      adcnBits_(conf.getParameter<uint32_t>("adcnBits")),
-      tdcsaturation_(conf.getParameter<double>("tdcsaturation")),
+    : linLSB_si_(conf.getParameter<double>("linLSB_si")),
+      adcsaturation_si_(conf.getParameter<double>("adcsaturation_si")),
+      tdcnBits_si_(conf.getParameter<uint32_t>("tdcnBits_si")),
+      tdcOnset_si_(conf.getParameter<double>("tdcOnset_si")),
+      adcnBits_si_(conf.getParameter<uint32_t>("adcnBits_si")),
+      tdcsaturation_si_(conf.getParameter<double>("tdcsaturation_si")),
+      linLSB_sc_(conf.getParameter<double>("linLSB_sc")),
+      adcsaturation_sc_(conf.getParameter<double>("adcsaturation_sc")),
+      tdcnBits_sc_(conf.getParameter<uint32_t>("tdcnBits_sc")),
+      tdcOnset_sc_(conf.getParameter<double>("tdcOnset_sc")),
+      adcnBits_sc_(conf.getParameter<uint32_t>("adcnBits_sc")),
+      tdcsaturation_sc_(conf.getParameter<double>("tdcsaturation_sc")),
       linnBits_(conf.getParameter<uint32_t>("linnBits")) {
-  adcLSB_ = adcsaturation_ / pow(2., adcnBits_);
-  tdcLSB_ = tdcsaturation_ / pow(2., tdcnBits_);
+  adcLSB_si_ = adcsaturation_si_ / pow(2., adcnBits_si_);
```

Comment:
@jbsauvan I seriously doubt this is computationally efficient wrt the division for the multiplication). Anyway it could be addressed separately as we are already late

Reply:
forget my comment, this is 2**adcnBits and it is called just in the constructor. Probably not the most efficient construct but not a real issue.

Reply:
Hi @fabiocos 
That's right.
I can improve that later. In any case it is just called in the constructor so the global impact on performance should be minimal.